modules/packages/UnrolledLinkedList.chpl

```diff
@@ -0,0 +1,1249 @@
+    }
+
+    pragma "no doc"
+    proc const ref this(i: int) const ref {
+      if boundsChecking && !_withinBounds(i) {
+        const msg = "Invalid unrolledLinkedList index: " + i:string;
+        halt(msg);
+      }
+
+      const ref result = _getRef(i);
+      return result;
+    }
+
+    /*
+      Iterate over the elements of this unrolledLinkedList.
+
+      :yields: A reference to one of the elements contained in this unrolledLinkedList.
+    */
+    iter these() ref {
+      var cur = _head;
+      while cur != nil {
+        for i in 0..#cur!.size {
+          yield cur!.data[i];
+        }
+        cur = cur!.next;
+      }
+    }
+
+    /*
+      Write the contents of this unrolledLinkedList to a channel.
+
+      :arg ch: A channel to write to.
+    */
+    proc writeThis(ch: channel) throws {
+      _enter();
+      
+      ch <~> "[";
+
+      var first = true;
+
+      for x in this {
+        if !first then ch <~> ", ";
+        else first = false;
+        ch <~> x;
+      }
+
+      ch <~> "]";
+
+      _leave();
+    }
+
+    /*
+      Returns `true` if this unrolledLinkedList contains zero elements.
+
+      :return: `true` if this unrolledLinkedList is empty.
+      :rtype: `bool`
+    */
+    proc const isEmpty(): bool {
+      return size == 0;
+    }
+
+    /*
+      The current number of elements contained in this unrolledLinkedList.
+    */
+    inline proc const size {
+      var result = 0;
+
+      on this {
+        _enter();
+        result = _size;
+        _leave();
+      }
+
+      return result;
+    }
+
+    /*
+      Returns the unrolledLinkedList's legal indices as the range ``0..<this.size``.
+
+      :return: ``0..<this.size``
+      :rtype: `range`
+    */
+    proc indices {
+      return 0..<this.size;
+    }
+
+    /*
+      Returns a new DefaultRectangular array containing a copy-initialized var of 
+      each of the elements contained in this unrolledLinkedList.
+
+      :return: A new DefaultRectangular array.
+    */
+    proc const toArray(): [] eltType {
+      if isNonNilableClass(eltType) && isOwnedClass(eltType) then
+        compilerError("toArray() method is not available on a 'unrolledLinkedList'",
+                      " with elements of a non-nilable owned type, here: ",
+                      eltType:string);
+
+
+      var result: [0..#_size] eltType;
+
+      on this {
+        _enter();
+
+        var i = 0;
+        for x in this {
+          result[i] = x;
+          i += 1;
+        }
+
+        _leave();
+      }
+
+      return result;
+    }
+
+  } // End record "unrolledLinkedList".
+
+  /*
+    Clear the contents of this unrolledLinkedList, then extend this now empty unrolledLinkedList with the
+    elements contained in another unrolledLinkedList.
+
+    .. warning::
+
+      This will invalidate any references to elements previously contained in
+      `lhs`.
+
+    :arg lhs: The unrolledLinkedList to assign to.
+    :arg rhs: The unrolledLinkedList to assign from. 
```

Comment:
Probably you can drop the args here.

Reply:
Right. And unfortunately (probably unsurprisingly, too) there is no guideline where I can point to.